File: rest-api/src/main/java/space/shefer/receipt/rest/entity/Merchant.java
package space.shefer.receipt.rest.entity;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "merchant")
@Data
public class Merchant {
  @Id
  @GeneratedValue
  @Column(name = "id")
  private Long id;
  @Column(name = "inn")
  private String inn;
  @Column(name = "name")
  private String name;
  @Column(name = "last_update_time")
  private Date lastUpdateTime;
  @Column(name = "address")
  private String address;
}

File: fns-sdk/src/main/kotlin/space/shefer/receipt/fnssdk/webclient/Merchant.kt
package space.shefer.receipt.fnssdk.webclient

import org.springframework.http.HttpEntity
import org.springframework.http.HttpMethod
import org.springframework.web.client.RestTemplate
import space.shefer.receipt.fnssdk.primainform.MerchantSuggestion
import java.net.URI

class Merchant {

    fun getMerchantFromPrimaInformOrNone(inn: String): MerchantSuggestion? {

        val responseEntity = RestTemplate().exchange(
                URI.create("https://www.prima-inform.ru/utils/search?query=" + inn),
                HttpMethod.GET,
                HttpEntity<String>(""),
                MerchantSuggestion::class.java
        )
        responseEntity.body?.suggestions?.
        filter {  it.data.inn == inn }?.
        map{suggestion-> mapOf("inn" to suggestion.data.inn, "address" to suggestion.data.address,"name" to suggestion.value)}

        return responseEntity.body
    }
}
